presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -161,6 +163,8 @@ public void rollback()
     {
         try {
             recordWriter.close(true);
+            // perform explicit delete of written file here as recordWriter.close() ignore the abort flag.
+            path.getFileSystem(conf).delete(path, false);
         }
         catch (IOException e) {
```

Comment:
What if `close()` succeeds and `delete()` throws `IOException`, do we still want to throw `HIVE_WRITER_CLOSE_ERROR`?